zonemaster/zonemaster-backend
# Uniform config property getter methods

## Context

This PR lays some groundwork for #685.
It also fixes a warning that was noted by @pnax in https://github.com/zonemaster/zonemaster-backend/pull/745#pullrequestreview-638608629.

## Scope

* No behavioral changes with regard to validation.
* No documentation changes to property getters.
* No code changes with regard to complex properties.
* New property getters are introduced, but the old ones are kept in place. Call sites aren't updated to use the new ones either. At first I wanted to include this cleanup in the PR, but the surface area was large enough that I'm leaving that for a follow-up PR.

## Changes

Format specification:
* Some details are included that were previously only documented in the POD of the getter methods.
* Simple properties undergo minor editing to make them more structurally similar to each other.

Implementation:
* Uniform getter methods are added for all simple properties.
* Defaults are added to all simple properties where it makes sense.
* All uniform getter method implementations are gathered up into a table-like section. IMHO the code is more readable this way as opposed to having each method placed next to its own POD. Pod::Coverage is still able to help us make sure all the methods are documented. Nothing interesting is going on in the method implementations anyway.
* The `SUBROUTINES` POD heading is replaced with `CONSTRUCTORS` and `METHODS`.

Extra:
* t/test_validate_syntax.t:
  1. The warning from https://github.com/zonemaster/zonemaster-backend/pull/745#pullrequestreview-638608629 is fixed
  2. The test now fails if it emits any warnings

## How to test this PR

1. Verify that t/confg.t contains unit tests for all default values in the format specification.
2. Verify that t/test_validate_syntax.t uses Test::NoWarnings.
